formatter/lib/elixir_test_runner/json_formatter.ex

```diff
@@ -0,0 +1,208 @@
+defmodule ElixirTestRunner.JSONFormatter do
+  @moduledoc false
+  use GenServer
+
+  import ExUnit.Formatter, only: [format_test_failure: 5]
+
+  # Callbacks
+
+  def init(opts) do
+    config = %{
+      seed: opts[:seed],
+      trace: opts[:trace],
+      colors: [enabled: false],
+      width: opts[:width] || 80,
+      slowest: opts[:slowest],
+      test_counter: %{},
+      test_timings: [],
+      failure_counter: 0,
+      skipped_counter: 0,
+      excluded_counter: 0,
+      invalid_counter: 0,
+
+      json_path: opts[:json_path] || false,
+      results: %{
+        status: :pass, # or :fail, or :error
+        message: nil,
+        tests: [
+          # {
+          #   name: "Test tthat the thing works",
+          #   status: :fail, # or :pass, or :error
+          #   message: "Expected 42 but got 123123"
+          # }
+        ]
+      }
+    }
+
+    {:ok, config}
+  end
+
+  def handle_cast({:suite_started, _opts}, config) do
+    {:noreply, config}
+  end
+
+  def handle_cast({:suite_finished, _run_us, _load_us}, config) do
+    json_results = "{\"errors\":#{Jason.encode!(config.results)}}"
```

Comment:
Might be personal preference, but I'd prefer to build the full map before encoding it, rather than building a string that way… Also I try to avoid banged functions when not in a pipe.

```ex
{:ok, json_results} = Jason.encode(%{errors: config.results})
```

Reply:
After seeing your example, I agree 100%.  Will change.

Reply:
change implemented, see new